src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -120,31 +122,52 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
   }
 
   async componentDidMount() {
-    const { store, dependentStores, isClusterScoped, tableId } = this.props;
-
-    if (this.canBeConfigured) this.hiddenColumnNames = new Set(userStore.preferences?.hiddenTableColumns?.[tableId]);
+    this.loadStores();
 
-    const stores = [store, ...dependentStores];
+    if (!this.props.isClusterScoped) {
+      disposeOnUnmount(this, [
+        namespaceStore.onContextChange(() => this.loadStores())
+      ]);
+    }
+  }
 
-    if (!isClusterScoped) stores.push(namespaceStore);
+  async componentWillUnmount() {
+    this.isUnmounting = true;
+    this.unsubscribeStores();
+  }
 
-    try {
-      stores.map(store => store.reset());
-      await Promise.all(stores.map(store => store.loadAll()));
-      const subscriptions = stores.map(store => store.subscribe());
+  @computed get stores() {
+    const { store, dependentStores, tableId } = this.props;
 
-      await when(() => this.isUnmounting);
-      subscriptions.forEach(dispose => dispose()); // unsubscribe all
-    } catch (error) {
-      console.log("catched", error);
+    if (this.canBeConfigured) {
+      this.hiddenColumnNames = new Set(userStore.preferences?.hiddenTableColumns?.[tableId]);
```

Comment:
This line raises an error on pod list when changing namespace filter.

Reply:
Thanks, will fix! 
P.s this is what happens when overusing `?.` :D cc @pashevskii